lib/Dialect/TTNN/IR/TTNNOpsAttrs.cpp

```diff
@@ -180,8 +180,8 @@ uint64_t TTNNLayoutAttr::getElementSizeBytes() const {
 // Example: memref<2x3!tt.tile<32x32xf32>> -> { 2, 3 }
 //
 // return The shape of the shard.
-llvm::SmallVector<int64_t> TTNNLayoutAttr::getShardShape() const {
```

Comment:
I think it requires to use a different constructor to make SmallVector to make an owning SmallVector, should be: 

::llvm::SmallVector<int64_t>(getMemref().getShape().begin(), getMemref().getShape().end());

We saw a similar issue before. https://github.com/tenstorrent/tt-mlir/pull/882/files

Reply:
I'm look in the source code `include/llvm/ADT/SmallVector.h:1232` and this doesn't seem like an issue.

Reply:
Ok you are right. In that case, I wonder the cause of this issue as it appears to be already making an owning copy?

Reply:
I've looked into that PR, difference is that `options.meshShape` is `ListOption`, but the same effect could've been achieved with  `implicitDeviceOptions.meshShape = ::llvm::SmallVector<int64_t>(*options.meshShape);` My guess is that the big part behind the decision that all `SmallVector` constructors are marked as explicit it exactly for the reason where you say "Yes, I know SmallVector will a acquire new memory and make a copy, and I agree to that."

Reply:
Ok thanks, yeah that was my question that how the changes in this PR are resolving the issue as the output for getShardShape() is already owning. But from your explanation it seems the issue propagated from `lib/Dialect/TTNN/Transforms/Passes.cpp:235`.